Realm/Realm/Exceptions/RealmExceptionCodes.cs

```diff
@@ -20,49 +20,179 @@ namespace Realms.Exceptions
+        RLM_ERR_DOMAIN_NOT_ALLOWED = 4303,
+        RLM_ERR_READ_SIZE_LIMIT_EXCEEDED = 4304,
+        RLM_ERR_INVALID_PARAMETER = 4305,
+        RLM_ERR_MISSING_PARAMETER = 4306,
+        RLM_ERR_TWILIO_ERROR = 4307,
+        RLM_ERR_GCM_ERROR = 4308,
+        RLM_ERR_HTTP_ERROR = 4309,
+        RLM_ERR_AWS_ERROR = 4310,
+        RLM_ERR_MONGODB_ERROR = 4311,
+        RLM_ERR_ARGUMENTS_NOT_ALLOWED = 4312,
+        RLM_ERR_FUNCTION_EXECUTION_ERROR = 4313,
+        RLM_ERR_NO_MATCHING_RULE = 4314,
+        RLM_ERR_INTERNAL_SERVER_ERROR = 4315,
+        RLM_ERR_AUTH_PROVIDER_NOT_FOUND = 4316,
+        RLM_ERR_AUTH_PROVIDER_ALREADY_EXISTS = 4317,
+        RLM_ERR_SERVICE_NOT_FOUND = 4318,
+        RLM_ERR_SERVICE_TYPE_NOT_FOUND = 4319,
+        RLM_ERR_SERVICE_ALREADY_EXISTS = 4320,
+        RLM_ERR_SERVICE_COMMAND_NOT_FOUND = 4321,
+        RLM_ERR_VALUE_NOT_FOUND = 4322,
+        RLM_ERR_VALUE_ALREADY_EXISTS = 4323,
+        RLM_ERR_VALUE_DUPLICATE_NAME = 4324,
+        RLM_ERR_FUNCTION_NOT_FOUND = 4325,
+        RLM_ERR_FUNCTION_ALREADY_EXISTS = 4326,
+        RLM_ERR_FUNCTION_DUPLICATE_NAME = 4327,
+        RLM_ERR_FUNCTION_SYNTAX_ERROR = 4328,
+        RLM_ERR_FUNCTION_INVALID = 4329,
+        RLM_ERR_INCOMING_WEBHOOK_NOT_FOUND = 4330,
+        RLM_ERR_INCOMING_WEBHOOK_ALREADY_EXISTS = 4331,
+        RLM_ERR_INCOMING_WEBHOOK_DUPLICATE_NAME = 4332,
+        RLM_ERR_RULE_NOT_FOUND = 4333,
+        RLM_ERR_API_KEY_NOT_FOUND = 4334,
+        RLM_ERR_RULE_ALREADY_EXISTS = 4335,
+        RLM_ERR_RULE_DUPLICATE_NAME = 4336,
+        RLM_ERR_AUTH_PROVIDER_DUPLICATE_NAME = 4337,
+        RLM_ERR_RESTRICTED_HOST = 4338,
+        RLM_ERR_API_KEY_ALREADY_EXISTS = 4339,
+        RLM_ERR_INCOMING_WEBHOOK_AUTH_FAILED = 4340,
+        RLM_ERR_EXECUTION_TIME_LIMIT_EXCEEDED = 4341,
+        RLM_ERR_NOT_CALLABLE = 4342,
+        RLM_ERR_USER_ALREADY_CONFIRMED = 4343,
+        RLM_ERR_USER_NOT_FOUND = 4344,
+        RLM_ERR_USER_DISABLED = 4345,
+        RLM_ERR_AUTH_ERROR = 4346,
+        RLM_ERR_BAD_REQUEST = 4347,
+        RLM_ERR_ACCOUNT_NAME_IN_USE = 4348,
+        RLM_ERR_INVALID_PASSWORD = 4349,
+        RLM_ERR_SCHEMA_VALIDATION_FAILED_WRITE = 4350,
+        RLM_ERR_APP_UNKNOWN = 4351,
+        RLM_ERR_MAINTENANCE_IN_PROGRESS = 4352,
+
+        RLM_ERR_WEBSOCKET_GOINGAWAY = 4400,
+        RLM_ERR_WEBSOCKET_PROTOCOLERROR = 4401,
+        RLM_ERR_WEBSOCKET_UNSUPPORTEDDATA = 4402,
+        RLM_ERR_WEBSOCKET_RESERVED = 4403,
+        RLM_ERR_WEBSOCKET_NOSTATUSRECEIVED = 4404,
+        RLM_ERR_WEBSOCKET_ABNORMALCLOSURE = 4405,
+        RLM_ERR_WEBSOCKET_INVALIDPAYLOADDATA = 4406,
+        RLM_ERR_WEBSOCKET_POLICYVIOLATION = 4407,
+        RLM_ERR_WEBSOCKET_MESSAGETOOBIG = 4408,
+        RLM_ERR_WEBSOCKET_INAVALIDEXTENSION = 4409,
+        RLM_ERR_WEBSOCKET_INTERNALSERVERERROR = 4410,
+        RLM_ERR_WEBSOCKET_TLSHANDSHAKEFAILED = 4411,
+
+        RLM_ERR_CALLBACK = 1000000,
+        RLM_ERR_UNKNOWN = 2000000,
+
+        RowDetached = 1000004000,
```

Comment:
Why this don't have the same casing?

Reply:
The `RLM_ERR...` ones are coming from Core. The PascalCased ones are custom codes that we define. I'll add a comment to make sure it's clear.